crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -60,20 +61,46 @@ impl<F: Send + Clone, Ctx: PaymentMethodRetrieve>
 
         // Stage 1
 
-        let payment_intent_fut = db
-            .find_payment_intent_by_payment_id_merchant_id(&payment_id, merchant_id, storage_scheme)
-            .map(|x| x.change_context(errors::ApiErrorResponse::PaymentNotFound));
+        let store = state.clone().store;
+        let m_merchant_id = merchant_id.clone();
+        let payment_intent_fut = tokio::spawn(
+            async move {
+                store
+                    .find_payment_intent_by_payment_id_merchant_id(
+                        &payment_id,
+                        m_merchant_id.as_str(),
+                        storage_scheme,
+                    )
+                    .map(|x| x.change_context(errors::ApiErrorResponse::PaymentNotFound))
+                    .await
+            }
+            .in_current_span(),
+        );
 
-        let mandate_details_fut = helpers::get_token_pm_type_mandate_details(
-            state,
-            request,
-            mandate_type.clone(),
-            merchant_account,
-            key_store,
+        let m_state = state.clone();
+        let m_mandate_type = mandate_type.clone();
+        let m_merchant_account = merchant_account.clone();
+        let m_request = request.clone();
+        let m_key_store = key_store.clone();
```

Comment:
Can we avoid some of the clones here?

Reply:
no we can't, the spawn needs a ownership of the value.